File: app/src/main/java/echo/com/surveys/dependencyInjection/DIComponent.java
package echo.com.surveys.dependencyInjection;

import dagger.Component;
import echo.com.surveys.activity.SurveyActivity;

import javax.inject.Singleton;

@Singleton
@Component(modules = {DIModule.class})
public interface DIComponent {
    void inject(SurveyActivity activity);
}

File: app/src/main/java/echo/com/surveys/dependencyInjection/DIComponent.kt
package echo.com.surveys.dependencyInjection

import dagger.Component
import echo.com.surveys.activity.SurveyActivity

import javax.inject.Singleton

@Singleton
@Component(modules = arrayOf(DIModule::class))
interface DIComponent {
    fun inject(activity: SurveyActivity)
}
